### PR TITLE
gateway/sia: Support proper {make,get}Bucket operations

### PR DESCRIPTION
**Pull request by harshavardhana**
## Description
Current implementation we faked the makeBucket operations
to allow for s3 clients to behave properly. But instead
we can create a placeholder zero byte file instead, which
is a hexadecimal representation of the bucket name itself.

## Motivation and Context
Support proper {make,get}Bucket operations

## How Has This Been Tested?
Manually using siad daemon

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added unit tests to cover my changes.
- [ ] I have added/updated functional tests in [mint](https://github.com/minio/mint). (If yes, add `mint` PR # here: )
- [x] All new and existing tests passed.

### CODE REVIEW

**Review by krisis — DISMISSED** [review]

File: cmd/gateway-sia.go [hunk 1]
```diff
@@ -350,11 +372,16 @@ func (s *siaObjects) ListObjects(bucket string, prefix string, marker string, de
...
 	root := s.RootDir + "/"
 
+	sha256sum := sha256.Sum256([]byte(bucket))
 	// FIXME(harsha) - No paginated output supported for Sia backend right now, only prefix
 	// based filtering. Once list renter files API supports paginated output we can support
 	// paginated results here as well - until then Listing is an expensive operation.
 	for _, sObj := range siaObjs {
 		name := strings.TrimPrefix(sObj.SiaPath, pathJoin(root, bucket, "/"))
+		// Skip the file created specially when bucket was created.
+		if name == hex.EncodeToString(sha256sum[:]) {
```
Comment (krisis) [inline cmd/gateway-sia.go:382]: Good catch!

**Review by krisis — DISMISSED** [review]

**Comment by harshavardhana** [comment]
@krisis @donatello please take a look again.

**Review by krisis — APPROVED** [review]

**Comment by nitisht** [comment]
moving reviewers since @donatello is not available currently

**Review by nitisht — APPROVED** [review]
LGTM as well

**Comment by nitisht** [comment]
ignoring codecov as this is a gateway PR

**Pull request merged**